cpp/log/cert.cc

```diff
@@ -1120,30 +1122,33 @@ Cert::Status TbsCertificate::DeleteExtension(int extension_nid) {
 
   // ExtensionIndex returns the first matching index - if the extension
   // occurs more than once, just give up.
-  int ignored;
-  status = ExtensionIndex(extension_nid, &ignored);
-  if (status == Cert::TRUE) {
+  status = ExtensionIndex(extension_nid);
+  if (status.ok() || status.status().CanonicalCode() == Code::INTERNAL) {
```

Comment:
Isn't this changing the logic? `INTERNAL` used to be `Cert::ERROR`, the condition used to be `status == Cert::TRUE`, and now it's the equivalent of `status == Cert::TRUE || status == Cert::ERROR)`?